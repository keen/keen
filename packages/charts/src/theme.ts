import { colors } from '@keen.io/colors';

import { Theme, Margins } from './types';

export const chartColors = [
  colors.lightBlue[500],
  colors.green[500],
  colors.orange[400],
  colors.blue[500],
  colors.yellow[500],
  colors.red[400],
  colors.purple[500],
  colors.green[200],
  colors.pink[500],
  colors.blue[200],
  colors.yellow[200],
  colors.purple[100],
  colors.lightBlue[100],
  colors.pink[200],
];

export const margins: Margins = { top: 50, right: 20, bottom: 50, left: 40 };

export const theme: Theme = {
  colors: chartColors,
  bar: {
    values: {
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 10,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.white['500'],
      },
    },
  },
  choropleth: {
    map: {
      stroke: colors.black['500'],
    },
    sphere: {
      enabled: false,
      backgroundColor: colors.gray['200'],
    },
    graticule: {
      enabled: false,
      color: colors.gray['400'],
    },
  },
  funnel: {
    header: {
      badge: {
        enabled: true,
        variant: 'black',
        typography: {
          fontStyle: 'normal',
          fontWeight: 'normal',
          fontSize: 14,
          fontFamily: 'Lato, sans-serif',
          fontColor: 'inherit',
        },
      },
      value: {
        enabled: true,
        typography: {
          fontStyle: 'normal',
          fontWeight: 'normal',
          fontSize: 20,
          fontFamily: 'Lato, sans-serif',
          fontColor: colors.blue['500'],
        },
      },
      title: {
        typography: {
          fontStyle: 'normal',
          fontWeight: 'normal',
          fontSize: 12,
          fontFamily: 'Lato, sans-serif',
          fontColor: colors.black['500'],
        },
      },
      backgroundColor: colors.gray['100'],
    },
    step: {
      backgroundColor: colors.gray['300'],
    },
  },
  metric: {
    value: {
      typography: {
        fontStyle: 'normal',
        fontWeight: 'lighter',
        fontSize: 60,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.blue['500'],
      },
    },
    prefix: {
      typography: {
        fontStyle: 'normal',
        fontWeight: 'lighter',
        fontSize: 20,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.blue['500'],
      },
    },
    suffix: {
      typography: {
        fontStyle: 'normal',
        fontWeight: 'lighter',
        fontSize: 20,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.blue['500'],
      },
    },
    icon: {
      enabled: false,
      position: 'top',
      style: 'solid',
      margins: { top: 0, left: 0, bottom: 0, right: 0 },
      type: 'brand',
    },
    excerpt: {
      icons: {
        increase: {
          color: colors.green['500'],
          type: 'arrow-up',
        },
        decrease: {
          color: colors.orange['500'],
          type: 'arrow-down',
        },
      },
      backgroundColor: colors.white['400'],
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 16,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.black['500'],
      },
    },
  },
  pie: {
    labels: {
      enabled: true,
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 12,
        fontFamily: 'Gangster Grotesk, sans-serif',
        fontColor: colors.black['500'],
      },
    },
  },
  gauge: {
    labels: {
      enabled: true,
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 14,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.black['500'],
      },
    },
    border: {
      backgroundColor: colors.blue['500'],
    },
    total: {
      enabled: true,
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 60,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.blue['400'],
      },
    },
  },
  donut: {
    labels: {
      enabled: true,
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 10,
        fontFamily: 'Gangster Grotesk, sans-serif',
        fontColor: colors.black['500'],
      },
    },
    total: {
      enabled: true,
      label: {
        typography: {
          fontStyle: 'normal',
          fontWeight: 'normal',
          fontSize: 14,
          fontFamily: 'Lato, sans-serif',
          fontColor: colors.black['400'],
        },
      },
      value: {
        typography: {
          fontStyle: 'normal',
          fontWeight: 'normal',
          fontSize: 24,
          fontFamily: 'Lato, sans-serif',
          fontColor: colors.blue['400'],
        },
      },
    },
  },
  table: {
    header: {
      typography: {
        fontStyle: 'normal',
        fontWeight: 'bold',
        fontSize: 14,
        fontFamily: 'Gangster Grotesk, sans-serif',
        fontColor: colors.white['500'],
        lineHeight: '17px',
      },
    },
    body: {
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 14,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.black['400'],
        lineHeight: '17px',
      },
    },
  },
  tooltip: {
    enabled: true,
    mode: 'light',
    labels: {
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 11,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.black['500'],
      },
    },
    values: {
      typography: {
        fontStyle: 'normal',
        fontWeight: 'bold',
        fontSize: 11,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.black['500'],
      },
    },
  },
  axisX: {
    enabled: true,
    padding: 5,
    tickSize: 10,
    tickPadding: 10,
    stroke: 1,
    color: colors.blue['100'],
    labels: {
      enabled: true,
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 10,
        fontFamily: 'Gangster Grotesk, sans-serif',
        fontColor: colors.black['500'],
      },
      radiusAngle: 0,
    },
    title: {
      alignment: 'center',
      padding: 20,
      typography: {
        fontStyle: 'normal',
        fontWeight: 'bold',
        fontSize: 14,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.blue['500'],
      },
    },
  },
  axisY: {
    enabled: true,
    padding: 5,
    tickSize: 0,
    tickPadding: 5,
    stroke: 0,
    color: colors.blue['100'],
    labels: {
      enabled: true,
      typography: {
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 10,
        fontFamily: 'Gangster Grotesk, sans-serif',
        fontColor: colors.black['500'],
      },
      radiusAngle: 0,
    },
    title: {
      alignment: 'center',
      padding: 20,
      typography: {
        fontStyle: 'normal',
        fontWeight: 'bold',
        fontSize: 14,
        fontFamily: 'Lato, sans-serif',
        fontColor: colors.blue['500'],
      },
    },
  },
  gridX: {
    enabled: true,
    color: colors.gray['400'],
  },
  gridY: {
    enabled: true,
    color: colors.gray['400'],
  },
  zeroIntersection: {
    color: colors.blue['500'],
  },
  hoverBar: {
    enabled: true,
    type: 'dark',
  },
};
