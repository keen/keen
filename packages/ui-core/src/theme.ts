/* eslint-disable @typescript-eslint/prefer-as-const */
import { colors } from '@keen.io/colors';
import { transparentize } from 'polished';
import { TextVariant, Typography } from './types';

export const textVariants: Record<TextVariant, Partial<Typography>> = {
  h1: {
    fontFamily: 'Gangster Grotesk',
    fontWeight: 'bold',
    fontSize: 36,
    lineHeight: '40px',
  },
  h2: {
    fontFamily: 'Gangster Grotesk',
    fontWeight: 'bold',
    fontSize: 30,
    lineHeight: '34px',
  },
  h3: {
    fontFamily: 'Gangster Grotesk',
    fontWeight: 'normal',
    fontSize: 20,
    lineHeight: '22px',
  },
  h4: {
    fontFamily: 'Lato',
    fontWeight: 'normal',
    fontSize: 16,
    lineHeight: '20px',
  },
  h5: {
    fontFamily: 'Lato',
    fontWeight: 'normal',
    fontSize: 12,
    lineHeight: '14px',
  },
  h6: {
    fontFamily: 'Lato',
    fontWeight: 'normal',
    fontSize: 10,
    lineHeight: '12px',
  },
  'body-normal': {
    fontFamily: 'Lato',
    fontWeight: 'normal',
    fontSize: 14,
    lineHeight: '16px',
  },
  'body-bold': {
    fontFamily: 'Lato',
    fontWeight: 'bold',
    fontSize: 14,
    lineHeight: '16px',
  },
  'number-xl': {
    fontFamily: 'Lato',
    fontWeight: 'lighter',
    fontSize: 55,
    lineHeight: '64px',
  },
  'number-l': {
    fontFamily: 'Lato',
    fontWeight: 'lighter',
    fontSize: 40,
    lineHeight: '50px',
  },
  'number-m': {
    fontFamily: 'Lato',
    fontWeight: 'lighter',
    fontSize: 30,
    lineHeight: '36px',
  },
  'number-s': {
    fontFamily: 'Lato',
    fontWeight: 'normal',
    fontSize: 16,
    lineHeight: '24px',
  },
};

export const theme = {
  font: {
    GangsterGrotesk: 'Gangster Grotesk',
    Lato: 'Lato',
  },
};

export const solidButtonVariants = {
  prop: 'variant',
  variants: {
    primary: {
      backgroundColor: colors.yellow['400'],
      borderColor: colors.yellow['400'],
      color: colors.black['500'],
      '&:focus': {
        boxShadow: `0 0 0 0.2rem rgba(218,165,80, .5)`,
      },
      '&:hover': {
        color: colors.black['400'],
        backgroundColor: colors.yellow['500'],
      },
    },
    secondary: {
      backgroundColor: colors.blue['500'],
      borderColor: colors.blue['500'],
      color: colors.white['500'],
      boxShadow: `0 2px 4px 0 ${transparentize(0.85, colors.black['500'])}`,
      '&:hover': {
        boxShadow: 'none',
        backgroundColor: colors.blue['400'],
      },
    },
    danger: {
      backgroundColor: colors.red['500'],
      borderColor: colors.red['500'],
      color: colors.white['500'],
      boxShadow: `0 1px 4px 0 ${transparentize(0.85, colors.black['500'])}`,
      '&:hover': {
        boxShadow: 'none',
        backgroundColor: colors.red['300'],
      },
    },
    success: {
      backgroundColor: colors.green['300'],
      borderColor: colors.green['300'],
      color: colors.white['500'],
      boxShadow: `0 1px 4px 0 ${transparentize(0.85, colors.black['500'])}`,
      '&:hover': {
        boxShadow: 'none',
        backgroundColor: colors.green['400'],
      },
    },
    blank: {
      backgroundColor: colors.white['500'],
      color: colors.blue['500'],
      boxShadow: `0 1px 4px 0 ${transparentize(0.85, colors.black['500'])}`,
      '&:hover': {
        boxShadow: 'none',
        backgroundColor: transparentize(0.8, colors.blue['100']),
      },
    },
  },
};
