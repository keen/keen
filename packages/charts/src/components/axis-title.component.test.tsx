import React from 'react';
import { render as rtlRender } from '@testing-library/react';

import AxisTitle from './axis-title.component';

const line = { x1: 0, x2: 100, y1: 100, y2: 0 };
const groupBox = { x: 10, y: 10, height: 10 };
const orientation = 'horizontal';
const title = 'Axis Title';

import { theme } from '../theme';

const render = (overProps: any = {}) => {
  const props = {
    line,
    groupBox,
    orientation,
    titleSettings: theme.axisX.title,
    ...overProps,
  };

  const wrapper = rtlRender(
    <svg>
      <AxisTitle {...props}>{title}</AxisTitle>
    </svg>
  );

  return {
    wrapper,
    props,
  };
};

test('should render AxisTitle', () => {
  const {
    wrapper: { container },
  } = render();
  expect(container).toMatchSnapshot();
});

test('should render AxisTitle text', () => {
  const {
    wrapper: { getByText },
  } = render();
  expect(getByText(title)).toBeInTheDocument();
});

test('should render horizontal title with theme props', () => {
  const {
    wrapper: { getByText },
  } = render();
  const text = getByText(title);
  expect(text).toMatchInlineSnapshot(`
    <text
      data-elementid="horizontal-axis-title"
      fill="#27566D"
      font-family="Lato, sans-serif"
      font-size="14"
      font-style="normal"
      font-weight="bold"
      text-anchor="middle"
      x="50"
      y="40"
    >
      Axis Title
    </text>
  `);
});

test('should render vertical title with theme props', () => {
  const {
    wrapper: { getByText },
  } = render({ orientation: 'vertical' });
  const text = getByText(title);
  expect(text).toMatchInlineSnapshot(`
    <text
      data-elementid="vertical-axis-title"
      fill="#27566D"
      font-family="Lato, sans-serif"
      font-size="14"
      font-style="normal"
      font-weight="bold"
      style="transform: rotate(-90deg);"
      text-anchor="middle"
      x="-50"
      y="-10"
    >
      Axis Title
    </text>
  `);
});
