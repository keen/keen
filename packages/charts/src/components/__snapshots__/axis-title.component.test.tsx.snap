// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AxisTitle 1`] = `
<div>
  <svg>
    <text
      data-elementid="horizontal-axis-title"
      fill="#27566D"
      font-family="Lato, sans-serif"
      font-size="14"
      font-style="normal"
      font-weight="bold"
      text-anchor="middle"
      x="50"
      y="40"
    >
      Axis Title
    </text>
  </svg>
</div>
`;
